### ISSUE TITLE
Exited with code 1

### ISSUE THREAD

**Issue by andy-1bg** [Codeception/robo-paracept#13]
I'm having an issue with ParallelExec and exit codes. Running the commands manually works correctly, but when ran through ParallelExec they fail because codeception exits with code 1 even though --no-exit is set. Any help would be much appreciated!

`[ParallelExec] ../vendor/bin/codecept run -c ../codeception.yml --group api_1 --no-exit api
[ParallelExec] ../vendor/bin/codecept run -c ../codeception.yml --group api_2 --no-exit api
[ParallelExec] '../vendor/bin/codecept run -c ../codeception.yml --group api_1 --no-exit api' exited with code 1
[ParallelExec]  Exit code 1  Time 13.654s`

**Comment by abdullah-jubayer** [comment]
@andy-1bg did you find out what was the reason for it? was it memory?

**Comment by javabudd** [comment]
I was not able to find a solution to the issue so instead of running a parallel execution, I grouped the tests and ran the them individually. 

`
THREAD_COUNT=12
for functionalGroup in $(seq 1 $THREAD_COUNT)
do
   sudo docker exec $(sudo docker ps |grep -w "webserver" | awk '{print $1}') /var/www/app/vendor/bin/codecept run -c /var/www/app/codeception.yml functional --fail-fast --no-ansi --group functional_$functionalGroup
done`

**Comment by abdullah-jubayer** [comment]
thankyou @javabudd

**Comment by jarrettj** [comment]
+1

**Comment by DavertMik** [comment]
yeah, I agree PHP is not the best way to do things in parallel. Use docker with Jenkins or GitLab CI to launch tests in multiple processes

**Comment by hadouke** [comment]
Hey guys. Can you give me hint - what 255 error mean?
`10:05:40  [ParallelExec]  '/var/lib/jenkins/workspace/forms-lead-test/vendor/bin/codecept run acceptance --group current_6 --html result_6.html' exited with code 255`

**Comment by ccsuperstar** [comment]
Hello everyone!
I have the same issue 
`[ParallelExec]  'vendor/bin/codecept run --env stage --override 'extensions: config: Codeception\Extension\RunFailed: fail-group: failed_4' --group parallel_4 --no-exit --xml report_4.xml --html report_4.html' exited with code 255 `
Have you any idea?

**Comment by reinholdfuereder** [comment]
Got exit code 255 also only when using parallel exec, because that ignores the `dir` of Codecept task (to change working directory) in "_robo/vendor/consolidation/robo/src/Task/Base/ParallelExec.php:94" by overriding it with the one from the Robo process:
`$this->processes[] = Process::fromShellCommandline($this->receiveCommand($command), getcwd());`
